### PR TITLE
Remove redundant files and folders in LISISO folder.

### PR DESCRIPTION
**Pull request by DivyaJyothiKasam**

### CODE REVIEW

**Comment by DivyaJyothiKasam** [comment]
################################Install scenario################################
[root@rhel75 LISISO]# ./install.sh
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS75
RPM's are available
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                          ########################################
Updating / installing...
kmod-microsoft-hyper-v-master.4.3.0.be########################################
microsoft-hyper-v-master.4.3.0.beta5-2########################################
Saving old initramfs
Installing new initramfs
Starting KVP Daemon....
Starting VSS Daemon....
Starting FCOPY Daemon....
microsoft-hyper-v-debuginfo-master.4.3########################################
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.
[root@rhel75 LISISO]# reboot
login as: root
root@10.128.76.246's password:
Last login: Mon Mar  4 03:27:34 2019 from lis-f2325.redmond.corp.microsoft.com
[root@rhel75 ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-862.el7.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta5
license:        GPL
retpoline:      Y
rhelversion:    7.5
srcversion:     CE2964B6A2652C90F0333C1
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-862.el7.x86_64 SMP mod_unload modversions
parm:           affinity_mode:vmbus channel cpu affinity mode: 0, 1 (int)

################################UnInstall scenario################################
[root@rhel75 LISISO]# ./uninstall.sh
Uninstalling the Linux Integration Services for Microsoft Hyper-V...
Removing Package..
Stopping KVP Daemon....
Stopping FCOPY Daemon....
Stopping VSS Daemon....
Create and install initramfs without Hyper-V drivers
Linux Integration Services for Hyper-V has been removed.  Please reboot your system.

Uninstalled Linux Integration Services for Microsoft Hyper-V...
[root@rhel75 LISISO]#
[root@rhel75 LISISO]# reboot
login as: root
root@10.128.76.246's password:
Last login: Mon Mar  4 03:31:32 2019 from lis-f2325.redmond.corp.microsoft.com
[root@rhel75 ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-862.el7.x86_64/kernel/drivers/hv/hv_vmbus.ko.xz
license:        GPL
retpoline:      Y
rhelversion:    7.5
srcversion:     FEB3E47F244407D2A8002E2
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
intree:         Y
vermagic:       3.10.0-862.el7.x86_64 SMP mod_unload modversions
signer:         Red Hat Enterprise Linux kernel signing key
sig_key:        51:73:02:3B:F8:16:37:D7:BF:3C:51:50:13:4E:EC:84:1B:96:FD:0B
sig_hashalgo:   sha256

################################Upgrade scenario################################
[root@rhel75 LISISO]# ./upgrade.sh
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS75
RPM's are available
Preparing...                          ########################################
Updating / installing...
kmod-microsoft-hyper-v-master.4.3.0.be########################################
microsoft-hyper-v-master.4.3.0.beta5-2########################################
Saving old initramfs
Installing new initramfs
Starting KVP Daemon....
Starting VSS Daemon....
Starting FCOPY Daemon....
microsoft-hyper-v-debuginfo-master.4.3########################################
Linux Integration Services for Hyper-V has been Upgraded. Please reboot your system
[root@rhel75 LISISO]# reboot
login as: root
root@10.128.76.246's password:
Last login: Mon Mar  4 03:35:58 2019 from lis-f2325.redmond.corp.microsoft.com
[root@rhel75 ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-862.el7.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta5
license:        GPL
retpoline:      Y
rhelversion:    7.5
srcversion:     CE2964B6A2652C90F0333C1
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-862.el7.x86_64 SMP mod_unload modversions
parm:           affinity_mode:vmbus channel cpu affinity mode: 0, 1 (int)
[root@rhel75 ~]#

File: LISISO/install.sh [hunk 1]
```diff
@@ -70,6 +70,20 @@ if [ $distro_version = "unknown" ]; then
     exit 1
 fi
 
+OracleDistroName=$(grep -ihs "Orac" /etc/oracle-release)
+OracleDistroVersion=[52,53,54,55,56,57,58,59,510,511,60,61,62,63]
+case $OracleDistroName in
```
Comment (prashantmhetar) [inline LISISO/install.sh]: Please perform unit testing for Oracle distro?
Also check if it can be tested for this negative condition

File: LISISO/upgrade.sh [hunk 2]
```diff
@@ -67,6 +67,20 @@ if [ $distro_version = "unknown" ]; then
     exit 1
 fi
 
+OracleDistroName=$(grep -ihs "Orac" /etc/oracle-release)
+OracleDistroVersion=[52,53,54,55,56,57,58,59,510,511,60,61,62,63]
+case $OracleDistroName in
```
Comment (prashantmhetar) [inline LISISO/upgrade.sh]: same as above for upgrade

File: LISISO/RPMS65/LIS52/install.sh [hunk 3]
```diff
@@ -40,7 +40,7 @@ if [ "$kmodrpm" != "" ] && [ "$msrpm" != ""  ]; then
        rpm -ivh $kmodrpm
        kmodexit=$?
        if [ "$kmodexit" == 0 ]; then
-              rpm -ivh --nodeps $msrpm
+              rpm -ivh $msrpm
```
Comment (prashantmhetar) [inline LISISO/RPMS65/LIS52/install.sh]: Do we need to remove --nodeps option?

Comment (prashantmhetar) [reply LISISO/RPMS65/LIS52/install.sh]: Seems copy is made wrongly from CentOS52 to RPMS65. Please check

File: LISISO/install.sh (hunk 1 above)
Comment (DivyaJyothiKasam) [reply LISISO/install.sh]: Tested for Oracle distro for negative condition.

Comment (DivyaJyothiKasam) [reply LISISO/install.sh]: [root@localhost LISISO]#./install.sh 
raji 58
installation is not supported for Oracle distros from series 52 to 63
[root@localhost LISISO]# ./upgrade.sh 
raji 59
Upgrade is not supoorted for Oracle distros frpm 52 to 63

File: LISISO/RPMS65/LIS52/install.sh (hunk 3 above)
Comment (DivyaJyothiKasam) [reply LISISO/RPMS65/LIS52/install.sh]: Done

File: LISISO/upgrade.sh (hunk 2 above)
Comment (DivyaJyothiKasam) [reply LISISO/upgrade.sh]: Done

Logs:
[root@localhost LISISO]# ./upgrade.sh
raji 59
Upgrade is not supoorted for Oracle distros frpm 52 to 63

**Comment by DivyaJyothiKasam** [comment]
Done unit testing for install, uninstall and upgrade scenarios for 
Centos58
Centos67
RHEL 68 to 610
RHEL 70 to 76
using the tar  "https://linuxpipelineshare.blob.core.windows.net/lis-rpm-builds/lis-builds/lis-rpm-master.4.3.0.beta5-2019-03-04.11.13.tar.gz"

**Comment by DivyaJyothiKasam** [comment]
Logs for 52 distro:
[root@localhost LISISO]# ./install.sh
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS58
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ########################################### [100%]
   1:kmod-microsoft-hyper-v ########################################### [100%]
Preparing...                ########################################### [100%]
   1:microsoft-hyper-v      ########################################### [ 50%]
Saving old initrd
Installing new initrd
        Done.
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
   2:microsoft-hyper-v-debug########################################### [100%]
 Linux Integration Services for Hyper-V has been installed. Please reboot your system
[root@localhost LISISO]# reboot

The system is going down for reboot NOW!
[root@localhost LISISO]#
login as: root
root@10.128.76.157's password:
Last login: Tue Mar  5 01:57:51 2019 from lis-f2325.redmond.corp.microsoft.com
[root@localhost ~]#
[root@localhost ~]# modinfo hv_vmbus
filename:       /lib/modules/2.6.18-308.el5/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master1
license:        GPL
srcversion:     E27F18E85305DF147CEF877
depends:
vermagic:       2.6.18-308.el5 SMP mod_unload gcc-4.1

File: cleanall.sh [hunk 4]
```diff
@@ -1,56 +1,13 @@
...
-rm -rf LISISO/RHEL511_UPDATE/lis-511/x86_64/*.rpm
-
+#cleanup LIS 5.0
 rm -rf LISISO/CentOS52/*.rpm
 rm -rf LISISO/CentOS52/lis-52/x86/*.rpm
 rm -rf LISISO/CentOS52/lis-52/x86_64/*.rpm
 
 rm -rf LISISO/CentOS53/*.rpm
 rm -rf LISISO/CentOS53/lis-53/x86/*.rpm
+<<<<<<< HEAD
```
Comment (iamshital) [inline cleanall.sh]: Unresolved conflicts. Please check all files.

**Comment by DivyaJyothiKasam** [comment]
Logs for Oracle distros:
########################INSTALL#########################
[root@localhost LISISO]# ./install.sh 
Installation is supported
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS71
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                          ################################# [100%]
Updating / installing...
   1:kmod-microsoft-hyper-v-master.4.3################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:microsoft-hyper-v-master.4.3.0.be################################# [ 50%]
Saving old initramfs
Installing new initramfs
Starting KVP Daemon....
Starting VSS Daemon....
Starting FCOPY Daemon....
   2:microsoft-hyper-v-debuginfo-maste################################# [100%]
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.
[root@localhost LISISO]# reboot

[root@localhost ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-229.el7.x86_64/weak-updates/mi                                                                       crosoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
rhelversion:    7.1
srcversion:     CE2964B6A2652C90F0333C1
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-229.20.1.el7.x86_64 SMP mod_unload modversi                                                                     
parm:           affinity_mode:vmbus channel cpu affinity mode: 0,                                                                        
[root@localhost ~]#

########################UNINSTALL#########################
[root@localhost ~]# ls
anaconda-ks.cfg
initial-setup-ks.cfg
[root@localhost ~]# ./uninstall.sh
[root@localhost ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-229.el7.x86_64/kernel/drivers/                                                                     
license:        GPL
rhelversion:    7.1
srcversion:     EE8D5AD1814D6675FE216D3
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
intree:         Y
vermagic:       3.10.0-229.el7.x86_64 SMP mod_unload modversions
signer:         Red Hat Enterprise Linux kernel signing key
sig_key:        59:2A:83:60:73:8A:89:6B:7C:53:5B:3B:0D:B8:AF:D5:B6                                                                       

########################UPGRADE#########################
[root@localhost LISISO]# ./upgrade.sh
Upgrade is supported
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS71
Upgrading the Linux Integration Services for Microsoft Hyper-V...
Preparing...                          ############################                                                                       ##### [100%]
Updating / installing...
   1:kmod-microsoft-hyper-v-master.4.3############################                                                                       ##### [ 33%]
   2:microsoft-hyper-v-master.4.3.0.be############################                                                                       ##### [ 67%]
Saving old initramfs
Installing new initramfs
Starting KVP Daemon....
Starting VSS Daemon....
Starting FCOPY Daemon....
   3:microsoft-hyper-v-debuginfo-maste############################                                                                       ##### [100%]
 Linux Integration Services for Hyper-V has been Upgraded. Please reboot your system
[root@localhost LISISO]# reboot
Connection to 10.128.77.17 closed by remote host.
Connection to 10.128.77.17 closed.
                                                                
[root@localhost ~]# modinfo hv_vmbus
filename:       /lib/modules/3.10.0-229.el7.x86_64/weak-updates/mi                                                                       crosoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
rhelversion:    7.1
srcversion:     CE2964B6A2652C90F0333C1
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-229.20.1.el7.x86_64 SMP mod_unload modversi                                                                      
parm:           affinity_mode:vmbus channel cpu affinity mode

**Comment by DivyaJyothiKasam** [comment]
Logs for Centos60:
##############INSTALL LOGS#######################
[root@centos60x64 LISISO]# ./install.sh 
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS60
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ########################################### [100%]
   1:kmod-microsoft-hyper-v ########################################### [100%]
Preparing...                ########################################### [100%]
   1:microsoft-hyper-v      ########################################### [ 50%]
Saving old initramfs
Installing new initramfs
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
   2:microsoft-hyper-v-debug########################################### [100%]
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.
[root@centos60x64 LISISO]# uname -r
2.6.32-71.el6.x86_64


[root@centos60x64 Desktop]# modinfo hv_vmbus
filename:       /lib/modules/2.6.32-71.el6.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
srcversion:     9E5FD2A85012F760628D808
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:        
vermagic:       2.6.32-71.el6.x86_64 SMP mod_unload modversions 

###############UNINSTALL############################
[root@centos60x64 LISISO]# ./uninstall.sh 
Uninstalling the Linux Integration Services for Microsoft Hyper-V...
Removing Package..
Stopping KVP Daemon....
Deleting KVP Daemon from Chkconfig....
Stopping VSS Daemon....
Deleting VSS Daemon from Chkconfig....
Stopping FCOPY Daemon....
Deleting FCOPY Daemon from Chkconfig....
Create and install initramfs without Hyper-V drivers
Linux Integration Services for Hyper-V has been removed.  Please reboot your system.
Uninstalled Linux Integration Services for Microsoft Hyper-V...

#############################UPGRADE########################
[root@centos60x64 LISISO]# ./upgrade.sh 
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS60
Upgrading the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ########################################### [100%]
   1:kmod-microsoft-hyper-v ########################################### [ 33%]
   2:microsoft-hyper-v      ########################################### [ 67%]
Saving old initramfs
Installing new initramfs
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
   3:microsoft-hyper-v-debug########################################### [100%]
 Linux Integration Services for Hyper-V has been Upgraded. Please reboot your system
[root@centos60x64 LISISO]# uanme -r
bash: uanme: command not found
[root@centos60x64 LISISO]# uname -r
2.6.32-71.el6.x86_64
[root@centos60x64 Desktop]# modinfo hv_vmbus
filename:       /lib/modules/2.6.32-71.el6.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
srcversion:     9E5FD2A85012F760628D808
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:        
vermagic:       2.6.32-71.el6.x86_64 SMP mod_unload modversions

File: LISISO/commonfunctions.sh [hunk 5]
```diff
@@ -7,27 +7,29 @@
...
-		*)
-			distro_name=unknown
-			return 1
-			;;
-	esac
-
+	OracleDistroName=$(grep -ihs "Oracle" /etc/oracle-release)
+	if [[ $OracleDistroName == *Oracle* ]]; then
+        	distro_name="Oracle"
+		distro_package_name=RPMS
```
Comment (prashantmhetar) [inline LISISO/commonfunctions.sh]: Do we need to set distro_package_name=RPMS in this function?

File: cleanall.sh (hunk 4 above)
Comment (DivyaJyothiKasam) [reply cleanall.sh]: Done

File: LISISO/commonfunctions.sh (hunk 5 above)
Comment (DivyaJyothiKasam) [reply LISISO/commonfunctions.sh]: yes. Then only we can enter in to corresponding RPMS67 or any  folder and install.

Comment (prashantmhetar) [reply LISISO/commonfunctions.sh]: if distro_package_name is set RPMS only , do we need to set it for different types of distros?

Comment (DivyaJyothiKasam) [reply LISISO/commonfunctions.sh]: we can place only once.

**Comment by DivyaJyothiKasam** [comment]
Logs for Centos63:
######################INSTALL_LOG###############################
[root@centos63x64 LISISO]# ./install.sh 
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS63
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ########################################### [100%]
   1:kmod-microsoft-hyper-v ########################################### [100%]
Preparing...                ########################################### [100%]
   1:microsoft-hyper-v      ########################################### [ 50%]
Saving old initramfs
Installing new initramfs
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
   2:microsoft-hyper-v-debug########################################### [100%]
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.

[root@centos63x64 Desktop]# modinfo hv_vmbus
filename:       /lib/modules/2.6.32-279.el6.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
srcversion:     9E5FD2A85012F760628D808
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:        
vermagic:       2.6.32-279.el6.
######################UNINSTALL_LOG###############################
root@centos63x64 Desktop]# cd /root/LISISO/
[root@centos63x64 LISISO]# ./uninstall.sh 
Uninstalling the Linux Integration Services for Microsoft Hyper-V...
Removing Package..
Stopping KVP Daemon....
Deleting KVP Daemon from Chkconfig....
Stopping VSS Daemon....
Deleting VSS Daemon from Chkconfig....
Stopping FCOPY Daemon....
Deleting FCOPY Daemon from Chkconfig....
Create and install initramfs without Hyper-V drivers
Linux Integration Services for Hyper-V has been removed.  Please reboot your system.
Uninstalled Linux Integration Services for Microsoft Hyper-V...

######################UPGRADE_LOG###############################
[root@centos63x64 Desktop]# cd /root/LISISO/
[root@centos63x64 LISISO]# ./upgrade.sh 
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS63
Upgrading the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ########################################### [100%]
   1:kmod-microsoft-hyper-v ########################################### [ 33%]
   2:microsoft-hyper-v      ########################################### [ 67%]
Saving old initramfs
Installing new initramfs
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
   3:microsoft-hyper-v-debug########################################### [100%]
 Linux Integration Services for Hyper-V has been Upgraded. Please reboot your system


[root@centos63x64 Desktop]# modinfo hv_vmbus
filename:       /lib/modules/2.6.32-279.el6.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        master.4.3.0.beta6
license:        GPL
srcversion:     9E5FD2A85012F760628D808
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:        
vermagic:       2.6.32-279.el6.x86_64 SMP mod_unload modversions

**Review by prashantmhetar — APPROVED** [review]
changes looks good

File: cleanall.sh [hunk 6]
```diff
@@ -1,287 +1,111 @@
...
-rm -rf LISISO/Oracle64/*.rpm
-rm -rf LISISO/Oracle64/update/*.rpm
-rm -rf LISISO/Oracle65/*.rpm
-rm -rf LISISO/Oracle66/*.rpm
-rm -rf LISISO/Oracle67/*.rpm
-rm -rf LISISO/Oracle68/*.rpm
-rm -rf LISISO/Oracle69/*.rpm
-
-
+#cleanup LIS 5.0
```
Comment (santoshx) [inline cleanall.sh]: need a space after #

File: cleanall.sh [hunk 7]
```diff
@@ -1,287 +1,111 @@
...
+rm -rf LISISO/RPMS52/*.rpm
+rm -rf LISISO/RPMS52/lis-52/x86/*.rpm
+rm -rf LISISO/RPMS52/lis-52/x86_64/*.rpm
```
Comment (santoshx) [inline cleanall.sh]: I think we don't need double directories.

**Review by santoshx — CHANGES_REQUESTED** [review]
Please write some details in commit message what is being done in the code.

**Review by santoshx — APPROVED** [review]
Looks okay overall.
